copy_project.py

```diff
@@ -0,0 +1,179 @@
+                    asset_url = self.asset_url.format(
+                        host=self.destination_url,
+                        project_id=self.destination_project_id,
+                        asset_type=key,
+                        asset_id=asset_id,
+                    )
+                    asset_relationship = post_to_destination(url=asset_url, body=None)
+                    if key in ["datasets", "artifacts"]:
+                        files = (
+                            asset_json["result_files"]
+                            if key == "simulation"
+                            else asset_json["file_names"]
+                        )
+                        source_download_url = self.download_url.format(
+                            host=self.source_url,
+                            asset_type=key,
+                            asset_id=asset,
+                        )
+                        destination_upload_url = self.upload_url.format(
+                            host=self.destination_url,
+                            asset_type=key,
+                            asset_id=asset_id,
+                        )
+                        for file in files:
+                            download_url_request = requests.get(
+                                url=source_download_url,
+                                params={"filename": file},
+                                timeout=120,
+                            )
+                            download_response = download_url_request.json()
+                            download_url = download_response["url"]
+                            download_file = requests.get(url=download_url, timeout=120)
+
+                            upload_url_request = requests.get(
+                                url=destination_upload_url,
+                                params={"filename": file},
+                                timeout=120,
+                            )
+                            upload_url = upload_url_request.json()
+
+                            upload_file = requests.put(
+                                url=upload_url["url"],
+                                files={f"{file}": download_file.content},
+                                timeout=120,
+                            )
+
+    def post_asset(self, asset, asset_type):
+        """
+        Function posts asset to destination.
+        """
+        post_url = self.post_url.format(host=self.destination_url, resource=asset_type)
+
+        return post_to_destination(url=post_url, body=scrub_obj(asset))
+
+
+if __name__ == "__main__":
+    # Initialize parser
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "-i", "--id", help="The id for the project in the source system."
+    )
+    parser.add_argument(
+        "-s", "--source", help="The source URL to copy the project from."
+    )
+    parser.add_argument(
+        "-d", "--destination", help="The destination URL to copy the project to."
+    )
```

Comment:
When asking for urls/host names, I find it best to put an example in the help text, otherwise I always end up looking in the code to see if I should include a trailing slash or even if it expects the https?:// at the beginning.